### ISSUE TITLE
How to change class weights in YOLOv8n to address class imbalance?

### ISSUE THREAD

**Issue by mmd-nemati** [ultralytics/ultralytics#15044]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I'm working on a cancer dataset which has ~54000 images. It has a huge imbalance; there are only 2% images with `cancer = True`. So it leads to poor performances on `cancer = True` class.

Since gathering more data is not possible, I tried to change the weights for each class in the training step, ```{cancer_zero = 1.0, cancer_one = 50.0}```, but I couldn't find the right argument or configuration for this.

```cls_weight``` or ```hyp``` or ```pos_weights``` parameters, as mentioned in various other issues like [this ](https://github.com/ultralytics/ultralytics/issues/2358) one are not available in the ```YOLO('yolov8n.pt')``` class or in the ```train``` function. 
Also editing ```data.yaml``` with adding the class weights to it didn't work.

So can you provide the correct way or parameter for the Model constructor or ```train``` function to modify the class weights?

Thanks!

**Comment by Y-T-G** [comment]
There's no official way to do this. You can try using callbacks to modify the loss function.

For example, for `Detect` model:

```python
import torch.nn as nn
from ultralytics import YOLO

model = YOLO("yolov8n.pt")
def add_weights(trainer):
    trainer.model.criterion.bce = nn.BCEWithLogitsLoss(reduction="none", pos_weight=[1, 50])
model.add_callback("on_train_start', add_weights)
model.train()
```

**Comment by mmd-nemati** [comment]
Hello @Y-T-G ,
Thank you so much for providing the code, I really appreciate it. Unfortunately, adjusting the class weights didn't significantly improve the performance, so I think I may need to explore other approaches to address this issue.

**Comment by Y-T-G** [comment]
Duplicate of https://github.com/ultralytics/ultralytics/issues/21082
